libraries/AP_Logger/AP_Logger_File.cpp

```diff
@@ -897,14 +884,13 @@ uint16_t AP_Logger_File::start_new_log(void)
         write_fd_semaphore.give();
         return 0xFFFF;
     }
-    hal.scheduler->expect_delay_ms(3000);
+    EXPECT_DELAY(hal, 3000);
```

Comment:
This is a behaviour change as the delay object will remain in scope well past the point we were zeroing the delay here.

This one may be worth creating an anonymous scope to give a tighter lifetime to the expected delay as it will definitely overlap with the other expected delays below.

Reply:
i'm quite happy with the wider scope